core/libdeno.rs

```diff
@@ -116,8 +117,13 @@ impl DenoIsolate {
     isolate.set_host_import_module_dynamically_callback(
       host_import_module_dynamically_callback,
     );
-    // TODO isolate_->SetData(0, this);
-    self.isolate_ = Some(isolate);
+    let self_box = unsafe {
+      let self_box = Box::new(self);
+      let self_ptr = Box::into_raw(self_box);
+      isolate.set_data(0, self_ptr as *mut c_void);
+      Box::from_raw(self_ptr)
+    };
+    self_box.isolate_ = Some(isolate);
```

Comment:
I don't know about this. I think we should return a Box<DenoIsolate> from the constructor and use Box::leak to get the pointer here.